src/mon/PGMonitor.cc

```diff
@@ -772,14 +772,14 @@ bool PGMonitor::prepare_pg_stats(MonOpRequestRef op)
     return false;
   }
 
-  last_osd_report[from] = ceph_clock_now();
-
   if (!stats->get_orig_source().is_osd() ||
       !mon->osdmon()->osdmap.is_up(from) ||
       stats->get_orig_source_inst() != mon->osdmon()->osdmap.get_inst(from)) {
     dout(1) << " ignoring stats from non-active osd." << dendl;
     return false;
   }
+      
+  last_osd_report[from] = ceph_clock_now();
```

Comment:
> In some cases

@wonzhq what is the case exactly? for example, 

1. an osd is marked down by monitor and then
1. we received a straying pg stat message from it
1. `last_osd_report` is marked with the time stamp.
1. after a while, a new osd joined in, and monitor assigned it the osd id of the previously marked down osd, // but the leader calls `check_osd_map()` after the map is committed, and `check_osd_map()` will clear the last_osd_report for that osd.
1. in the `tick()` the newly joined osd could be wrongly mark down? (but i doubt. see above)